src/Components/Modal.js

```diff
@@ -1,28 +1,49 @@
-import { useEffect } from 'react';
-import StarRating from './StarRating'
+import { useEffect } from "react";
+import StarRating from "./StarRating";
 
-import './Modal.scss';
+import "./Modal.scss";
+
+import YoutubeEmbedVideo from "youtube-embed-video";
 
 function Modal({
   selectedMovieID,
   setModalIsOpen,
   selectedMovieObj,
   setSelectedMovieObj,
+  selectedMovieTrailerLink,
+  setSelectedMovieTrailerLink,
 }) {
   function getSingleMovieApi() {
     fetch(
-      `https://rancid-tomatillos.herokuapp.com/api/v2//movies/${selectedMovieID}`,
+      `https://rancid-tomatillos.herokuapp.com/api/v2//movies/${selectedMovieID}`
     )
-      .then(res => res.json())
-      .then(data => {
+      .then((res) => res.json())
+      .then((data) => {
         setSelectedMovieObj(data);
         console.log(data);
       })
-      .catch(err => console.log(err));
+      .catch((err) => console.log(err));
+  }
+
+  function getSingleMovieVideoApi() {
+    fetch(
+      `https://rancid-tomatillos.herokuapp.com/api/v2/movies/${selectedMovieID}/videos`
+    )
+      .then((res) => res.json())
+      .then((data) => {
+        const trailer = data.videos.find((video) => {
+          return video.type === "Trailer";
+        });
+        console.log(trailer);
+        const trailerKey = trailer.key;
```

Comment:
This variable assignment might not be necessary - just pass `trailer.key` directly into the method below if you don't need this variable for more than one thing!